src/main/scala/ciris/aws/ssm/package.scala

```diff
@@ -1,81 +1,50 @@
 package ciris.aws
 
+import cats.effect.Blocker
 import ciris._
-import ciris.api._
 import com.amazonaws.auth._
 import com.amazonaws.regions.Regions
-import com.amazonaws.services.simplesystemsmanagement.model.{
-  GetParameterRequest,
-  ParameterNotFoundException
-}
-import com.amazonaws.services.simplesystemsmanagement.AWSSimpleSystemsManagementClientBuilder
-
-import scala.util.{Try, Success}
+import com.amazonaws.services.simplesystemsmanagement._
+import com.amazonaws.services.simplesystemsmanagement.model._
 
 package object ssm {
+  final def param(key: String, blocker: Blocker)(
+    implicit region: Regions = Regions.EU_WEST_1,
+    credsProvider: AWSCredentialsProvider = new DefaultAWSCredentialsProviderChain()
+  ): ConfigValue[String] =
+    ConfigValue.blockOn(blocker) {
+      ConfigValue.suspend {
+        val ssmClient =
+          AWSSimpleSystemsManagementClientBuilder
+            .standard()
+            .withRegion(region)
+            .withCredentials(credsProvider)
+            .build()
+
+        val configKey =
+          ConfigKey(s"parameter $key from AWS SSM")
 
-  val SecretString: ConfigKeyType[String] =
-    ConfigKeyType[String]("secret string from AWS parameter store")
+        try {
+          val result =
+            ssmClient.getParameter {
+              new GetParameterRequest()
+                .withWithDecryption(true)
+                .withName(key)
+            }
 
-  def awsSsmSource(
-      region: Regions,
-      credsProvider: AWSCredentialsProvider
-  ): ConfigSource[Id, String, String] =
-    ConfigSource.fromTryOption(SecretString) { key =>
-      val ssmClient = AWSSimpleSystemsManagementClientBuilder
-        .standard()
-        .withRegion(region)
-        .withCredentials(credsProvider)
-        .build()
-      try {
-        val request = new GetParameterRequest()
-          .withName(key)
-          .withWithDecryption(true)
-        Try(ssmClient.getParameter(request))
-          .map { result =>
+          val loaded =
             for {
-              param <- Option(result.getParameter)
-              value <- Option(param.getValue)
-            } yield value
-          }
-          .recover {
-            case _: ParameterNotFoundException => None
-          }
-      } finally {
-        ssmClient.shutdown()
+              parameter <- Option(result.getParameter)
+              value <- Option(parameter.getValue)
+            } yield ConfigValue.loaded(configKey, value)
+
+          loaded.getOrElse(ConfigValue.missing(configKey))
+        } catch {
+          case _: ParameterNotFoundException =>
+            ConfigValue.missing(configKey)
+        } finally {
+          ssmClient.shutdown()
```

Comment:
Now we have cats-effect at our disposal we could do some Resource jazz with the SSM client instead of using try-finally? I don't have a strong opinion either way though.

Reply:
Yes, I actually started doing just that, but then wanted to get your thoughts. The client can be shared for multiple parameters, but it will mean a two-step process, e.g. `params(blocker).use { param => ... }`. If that sounds good, happy to change it.

Reply:
I pushed https://github.com/ovotech/ciris-aws-ssm/pull/8/commits/139d3ffa887392f0300be4b024e302a6b448f043 with this change, so you can have a look.

Reply:
👍 I like this. Re-using the client will speed things up slightly, and only having to pass the blocker once is more ergonomic.

Reply:
Agreed. Let's keep it.